test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1

```diff
@@ -163,5 +163,70 @@ Describe "Get-Content" -Tags "CI" {
   ) {
     param($cmdlet)
     (Get-Command $cmdlet).Parameters["stream"] | Should BeNullOrEmpty
-  }  
+  }
+ 
+  It "Should return no content when an empty path is used with -Raw switch" {
+    Get-ChildItem $TestDrive -Filter "*.raw" | Get-Content -Raw | Should Be $null
+  }
+
+  It "Should return no content when -TotalCount value is 0" {
+	  Get-Content -Path $testPath -TotalCount 0 | Should Be $null
+  }
+
+  It "Should throw TailAndHeadCannotCoexist when both -Tail and -TotalCount are used" {
+    { 
+      Get-Content -Path $testPath -Tail 1 -TotalCount 1 -ErrorAction Stop
+    } | ShouldBeErrorId "TailAndHeadCannotCoexist,Microsoft.PowerShell.Commands.GetContentCommand"
+  }
+
+  It "Should throw TailNotSupported when -Tail used with an unsupported provider" {
+    Push-Location env:
+    {
+      Get-Content PATH -Tail 1 -ErrorAction Stop
+    } | ShouldBeErrorId "TailNotSupported,Microsoft.PowerShell.Commands.GetContentCommand"
+  	Pop-Location
+  }
+
+  It "Should throw InvalidOperation when -Tail and -Raw are used" {
+    {
+      Get-Content -Path $testPath -Tail 1 -ErrorAction Stop -Raw
+    } | ShouldBeErrorId "InvalidOperation,Microsoft.PowerShell.Commands.GetContentCommand"
+  }
+
+  It "Should Get-Content containing multi-byte chars for a variety of -Tail and -ReadCount values" {
+    $firstLine = "Hello,World"
+    $secondLine = "Hello2,World2"
+    $thirdLine = "Hello3,World3"
+    $fourthLine = "Hello4,World4"
+    $fileContent = $firstLine,$secondLine,$thirdLine,$fourthLine
```

Comment:
I'd like to see this test refactored into a Context and each set of tests become a separate It within the Context. There are too many Should checks within this It.

Something like:
```powershell
Context "Check Get-Content containing multi-byte chars" {
    BeforeAll {
        $firstLine = "Hello,World" 
        $secondLine = "Hello2,World2" 
        $thirdLine = "Hello3,World3" 
        $fourthLine = "Hello4,World4" 
        $fileContent = $firstLine,$secondLine,$thirdLine,$fourthLine 
    }
    It "Some descriptive title" {
        Set-content -Path $testPath $fileContent 
        $result = Get-Content -Path $testPath -ReadCount -1 -Tail 5 -Encoding UTF7 
        $result.Length | Should Be 4
    }
    <Additional It blocks, one for each group of checks>
} 
```